### PR TITLE
Issue #227: make clear that compatibility is just for on-the-wire format

### PR DESCRIPTION
**Pull request by ePaul**
This is the second half of #227 (the first one is #241).

### CODE REVIEW

File: compatibility/Compatibility.md [hunk 1]
```diff
@@ -13,8 +13,17 @@ There are two techniques to change APIs without breaking them:
 - introduce new API versions and still support older versions
 
 We strongly encourage using compatible API extensions and discourage versioning.
-With Postel’s Law in mind, here are some rules for providers and consumers that
-allow us to make compatible changes without versioning:
+With Postel’s Law in mind, the following some rules for providers and consumers
```
Comment (whiskeysierra) [inline compatibility/Compatibility.md]: > the following rules

File: compatibility/Compatibility.md [hunk 2]
```diff
@@ -13,8 +13,17 @@ There are two techniques to change APIs without breaking them:
...
+allow us to make compatible changes without versioning.
+
+Please note that the compatibility guarantees are for the on-the-wire format,
+i.e. they ensure that consumers relying on the old API definition will work
+with servers which already implement a newer API definitions, assuming both API
+designers and client and server implementors follow the following rules.
```
Comment (whiskeysierra) [inline compatibility/Compatibility.md]: > adhere to the following rules?

File: compatibility/Compatibility.md (hunk 1 above)
Comment (ePaul) [inline compatibility/Compatibility.md]: Hmm, I wanted to explain that I mean just some rules, not all of them. Though I guess then I should have gone and counted (`the following 5 rules`).

File: compatibility/Compatibility.md (hunk 2 above)
Comment (ePaul) [inline compatibility/Compatibility.md]: Updated.

File: compatibility/Compatibility.md [hunk 3]
```diff
@@ -13,8 +13,17 @@ There are two techniques to change APIs without breaking them:
...
-allow us to make compatible changes without versioning:
+With Postel’s Law in mind, the following five rules for providers and consumers
+allow us to make compatible changes without versioning.
+
+Please note that the compatibility guarantees are for the on-the-wire format,
+i.e. they ensure that consumers relying on the old API definition will work
+with servers which already implement a newer API definitions, assuming both API
+designers and client and server implementors adhere to the following rules.
+
+(Binary or source compatibility of code generated from an API definition is not covered
```
Comment (tkrop) [inline compatibility/Compatibility.md]: I would express this less reluctant:

**Note:** compatibility of code generated from an API definition is not covered by the rules. If you update the definition, it has to be expected that code changes are necessary.

File: compatibility/Compatibility.md [hunk 4]
```diff
@@ -13,8 +13,17 @@ There are two techniques to change APIs without breaking them:
...
 We strongly encourage using compatible API extensions and discourage versioning.
-With Postel’s Law in mind, here are some rules for providers and consumers that
-allow us to make compatible changes without versioning:
+With Postel’s Law in mind, the following five rules for providers and consumers
+allow us to make compatible changes without versioning.
+
+Please note that the compatibility guarantees are for the on-the-wire format,
+i.e. they ensure that consumers relying on the old API definition will work
+with servers which already implement a newer API definitions, assuming both API
+designers and client and server implementors adhere to the following rules.
```
Comment (dehora) [inline compatibility/Compatibility.md]: > +with servers which already implement a newer API definitions,

Can remove the plural here - 

"""
with servers which already implement a newer API definition, 
"""

File: compatibility/Compatibility.md [hunk 5]
```diff
@@ -13,8 +13,17 @@ There are two techniques to change APIs without breaking them:
...
+
+(Binary or source compatibility of code generated from an API definition is not covered
+by this – if you update the definition to a new version, it is expected that code changes
+are necessary.)
```
Comment (dehora) [inline compatibility/Compatibility.md]: I don't think the parens here are needed, it's a constraint on API clients and can be phrased as a specification - 

"""
Binary or source compatibility of code generated from an API definition is not covered by this – API designers MAY update their Open API definition to a new version such that code changes by client generators would be necessary, as long as the runtime representations remain compatible.
"""

File: compatibility/Compatibility.md (hunk 4 above)
Comment (ePaul) [inline compatibility/Compatibility.md]: Done.

File: compatibility/Compatibility.md (hunk 3 above)
Comment (ePaul) [inline compatibility/Compatibility.md]: Done (slightly adapted).

File: compatibility/Compatibility.md (hunk 5 above)
Comment (ePaul) [inline compatibility/Compatibility.md]: I think even more that API designers should not care about what client code generators are doing with this, as long as they adhere to the rules here.

I removed the parentheses, and reworded it a bit, mostly following the proposal of @tkrop.

**Comment by ePaul** [comment]
:+1:

**Comment by tkrop** [comment]
👍

**Pull request merged**